### PR TITLE
Changes to release workflow

### PR DESCRIPTION
**Pull request by cdivitotawela**
## PR description
- Workflow environment variable changed to RELEASE_VERSION instead of RELEASE_NAME to make it more meaningful

- Workflows draft-release and docker-promote previously trigger workflow container-verify. This can result in trigger is successful but actual verification workflow fails. This will not be reflected in the draft-release and docker-promote workflows. Container verify code is embedded in the draft-release and docker-promote workflows to avoid this confusion

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: .github/workflows/docker-promote.yml [hunk 1]
```diff
@@ -58,24 +58,49 @@ jobs:
...
+            runner: ubuntu-22.04
+          - tag: latest
+            platform: ''
+            runner: ubuntu-22.04
+          - tag: ${{ needs.validate.outputs.release_version }}-arm64
+            platform: ''
+            runner: besu-arm64
     steps:
       - name: Checkout
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
```
Comment (siladu) [inline .github/workflows/docker-promote.yml:90]: Not directly relevant to your changes but wondering if we really need to checkout the whole repo here? 
Is it just to get the script? I assume it pulls from our dockerhub so doesn't need the source to verify. Wonder if we can get hold of just the script instead.

File: .github/workflows/docker-promote.yml [hunk 2]
```diff
@@ -58,24 +58,49 @@ jobs:
...
+          - tag: ${{ needs.validate.outputs.release_version }}
+            platform: ''
+            runner: ubuntu-22.04
+          - tag: ${{ needs.validate.outputs.release_version }}-amd64
+            platform: 'linux/amd64'
+            runner: ubuntu-22.04
+          - tag: latest
+            platform: ''
+            runner: ubuntu-22.04
+          - tag: ${{ needs.validate.outputs.release_version }}-arm64
```
Comment (siladu) [inline .github/workflows/docker-promote.yml:85]: Haven't we already checked the release_version variants in draft-release?

Think we should check latest, latest-amd64, latest-arm64 here

**Review by siladu — APPROVED** [review]
PR looks sound, my comments are beyond what the PR was intended for - up to you if you want to merge this one first.

File: .github/workflows/docker-promote.yml (hunk 1 above)
Comment (cdivitotawela) [reply .github/workflows/docker-promote.yml:90]: Parse checkout is possible. Wonder it is slower than the full checkout. Let me check this and switch to parse checkout only the file.

Comment (cdivitotawela) [reply .github/workflows/docker-promote.yml:90]: Looks like parse checkout is faster. Let me update this.

File: .github/workflows/docker-promote.yml (hunk 2 above)
Comment (cdivitotawela) [reply .github/workflows/docker-promote.yml:85]: Good point. I will add latest-amd64 and latest-arm64

File: .github/workflows/docker-promote.yml (hunk 1 above)
Comment (cdivitotawela) [reply .github/workflows/docker-promote.yml:91]: Updated

File: .github/workflows/BesuContainerVerify.sh [hunk 3]
```diff
@@ -57,7 +57,7 @@ else
 fi
 
 # For the latest tag check the version match
-if [[ ${TAG} == "latest" && ${CHECK_LATEST} == "true" ]]
+if [[ ${TAG} =~ "latest" && ${CHECK_LATEST} == "true" ]]
```
Comment (siladu) [inline .github/workflows/BesuContainerVerify.sh:60]: nit, could tighten up the regex slightly: 
```suggestion
if [[ ${TAG} =~ "^latest" && ${CHECK_LATEST} == "true" ]]
```

Comment (cdivitotawela) [reply .github/workflows/BesuContainerVerify.sh:60]: I tried that did not work. Let me check again whether I missed something

Comment (cdivitotawela) [reply .github/workflows/BesuContainerVerify.sh:60]: Found it. I should not use quotes around the pattern. Will fix it.

Comment (cdivitotawela) [reply .github/workflows/BesuContainerVerify.sh]: Updated

**Pull request merged**